catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SortedQueryMonitor.java

```diff
@@ -0,0 +1,219 @@
+
+        if (sortBy != null && sortBy.getPropertyName() != null) {
+            PropertyName sortingProp = sortBy.getPropertyName();
+            String sortType = sortingProp.getPropertyName();
+            SortOrder sortOrder = (sortBy.getSortOrder() == null) ?
+                    SortOrder.DESCENDING :
+                    sortBy.getSortOrder();
+            logger.debug("Sorting type: {}", sortType);
+            logger.debug("Sorting order: {}", sortBy.getSortOrder());
+
+            // Temporal searches are currently sorted by the effective time
+            if (Metacard.EFFECTIVE.equals(sortType) || Result.TEMPORAL.equals(sortType)) {
+                coreComparator = new TemporalResultComparator(sortOrder);
+            } else if (Result.DISTANCE.equals(sortType)) {
+                coreComparator = new DistanceResultComparator(sortOrder);
+            } else if (Result.RELEVANCE.equals(sortType)) {
+                coreComparator = new RelevanceResultComparator(sortOrder);
+            }
+        }
+
+        List<Result> resultList = new ArrayList<>();
+        long totalHits = 0;
+        Set<ProcessingDetails> processingDetails = returnResults.getProcessingDetails();
+
+        Map<String, Serializable> returnProperties = returnResults.getProperties();
+
+        for (int i = futures.size(); i > 0; i--) {
+            String sourceId = "Unknown Source";
+            Source source = null;
+            try {
+                Future<SourceResponse> future;
+                if (query.getTimeoutMillis() < 1) {
+                    future = completionService.take();
+                } else {
+                    future = completionService
+                            .poll(getTimeRemaining(deadline), TimeUnit.MILLISECONDS);
+                    if (future == null) {
+                        timeoutRemainingSources(processingDetails);
+                        break;
+                    }
+                }
+
+                source = futures.remove(future);
+                if (source != null) {
+                    sourceId = source.getId();
+                }
+
+                SourceResponse sourceResponse = future.get();
+
+                if (sourceResponse == null) {
+                    logger.info("Source {} returned null response", sourceId);
+                    processingDetails
+                            .add(new ProcessingDetailsImpl(sourceId, new NullPointerException()));
+                } else {
+                    resultList.addAll(sourceResponse.getResults());
+                    totalHits += sourceResponse.getHits();
+
+                    Map<String, Serializable> properties = sourceResponse.getProperties();
+                    returnProperties.putAll(properties);
+                }
+            } catch (InterruptedException e) {
+                if (source != null) {
+                    // First, add interrupted processing detail for this source
```

Comment:
Other than the deadline calculation, this was the only change of note in this extracted class. We weren't adding the processing detail for the actual source that was interrupted, only the ones remaining.